packages/text-annotator/src/SelectionHandler.ts

```diff
@@ -193,6 +193,10 @@ export const SelectionHandler = (
 
     // Logic for selecting an existing annotation
     const clickSelect = () => {
+      if (selection.userSelectAction === UserSelectAction.NONE) {
+        return;
+      }
+
```

Comment:
That logic is already covered in the parent `@annotorious/annotorious` package. Even though the `selection.userSelection` will be invoked, the actual selection state will always remain empty. [See this source](https://github.com/annotorious/annotorious/blob/03ee23fa905a3fc448b4aa51d769c15ccb31c93a/packages/annotorious-core/src/state/Selection.ts#L69-L79)

Reply:
Also, we cannot compare the `userSelectAction` with the enum value directly, because the `userSelectAction` can also be an expression: `((a: T) => UserSelectAction)`

Reply:
Yes - what I'm trying to achieve is to control the selection myself, described in #153

Reply:
I'll work on the `userSelectAction` expression, thanks! I just realised this - sorry, am pretty new here.

Reply:
Good point. Yes, indeed the core package already checks this against the current `userSelectionAction` (value or function).

Reply:
> That logic is already covered in the parent @annotorious/annotorious package.

The logic in the parent package seems to set the selection to an empty array if `userSelectAction` is `UserSelectAction.NONE`. Is this intended behaviour?

Reply:
Yes that's as intended. The three settings are:

- `EDIT` - a hint to the UI that selecting should make the annotation shape editable. (Only supported for image annotations so far.)
- `SELECT` - should make selected annotations surface in the selection event, but not make it editable.
- `NONE` - should make annotations inert. Clicking does nothing, not even fire selection events.

Reply:
`NONE` right now doesn't make annotations inert though. It removes programmatic selections

Reply:
Yes, I'm trying to do multi-selection behaviour. The selection API does support a list of annotations, but I have to set `userSelectAction` to `NONE`. And when I do that, clicking on any annotation will clear the list of annotations that my program has selected

Reply:
> I guess - my expectation of inertness is that this default clause does nothing.

I guess that's something that might be worth discussing. In principle, my view is: clicking an inert annotation would be considered the same as clicking on empty space, which is supposed to clear the selection. What's __not__ covered in the current code, of course, is that you might click an inert annotation that overlaps a non-inert one. In this case, I'd expect the non-inert one to get selected (and the inert one to get ignored). The current code, however, would simply clear the selection as you say.